### DISCUSSION TITLE
General question about organizing categories.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#290] [Help]
I have just a general question regarding categories and how we should add them to products.  As an example, we have the product "music bingo" which has the primary category of `bingo` however when I add this through the api it doesn't seem to work the way I intend.

I'm setting it through the api as the `category_id` and I'm also adding it first with all the other secondary categories in the `category_index` but when I look at our swell storefront it doesn't show up https://beta.cirqus.co/categories/bingo and also in the admin area is shows as the middle most category (see image).

Through the API I can see it setup like so:

```
    "category_id"=>"64354e81e50b3e0012b30c4b", # this is the ID for bingo
    "category_index"=>
     {"sort"=>{"64354e81e50b3e0012b30c4b"=>0}, # bingo
      "id"=>
       ["64354e81e50b3e0012b30c4b", # bingo
        "64354e7a8e12ec0012c926f0",
        "64354e8a387a320012a315d3", 
        "64354e5f73d5db0012585bd0", # <--- this one it chooses, why this one???
        "64354e86b7b27d001113b048"]},
```

![CleanShot 2023-06-29 at 22 28 16](https://github.com/swellstores/community/assets/48241/198a92d1-4992-4106-8b06-c7512a3c01c9)

What can I do to make sure I can control the primary category?

**Comment by logeshswell** [comment]
@holden I guess the ordering of categories in the **Product detail** page depends on the order in which the categories are sorted in the **Categories** page.
